design/comment-scanning.md

```diff
@@ -0,0 +1,110 @@
+# Polymer Analyzer - Comment Directives & Comment Annotations
+> Author @FredKSchott
+> Status: Draft
+> Last Update: 2017-04-25
+
+## Objective
+- To create a paradigm for formatting document metadata and instructions for analysis.
+- Immediately driving this discussion is the need to support polymer-linter directives which can enable/disable lint rules within entire documents and/or single sections of documents.
+
+## Goals
+- Allow the analyzer to surface the information contained within polymer-lint directives so that the linter can properly report & ignore warnings.
+- Create a more general definition for a directive that is easy-to-use for code authors and fits into the data model of the analyzer.
+- Create a more general definition for metadata (ex: @polymerElement) that is attached to an already-existing feature.
```

Comment:
Please see changes, I've gone into more detail about Annotations and how they differ from directives. 

I don't defining a range as cumbersome compared to the uncertainty with how an attached directive would behave. For example:
- If it is implicit whether a directive is attached or standalone, how do we communicate that to the user and not give them a 👟🔫? 
- What if you want to define an unattached directive above a feature? It would always be "attached" by the AST parser.
- What if an unattached directive ("lint enable") exists inside of a feature with an attached directive ("lint disable")? Where is linting enabled and where is it disabled? How do they interact?

If you desire for "attached" directives is convenience, I'd recommend we kick that until after we've had some users play with standalone directives.

(As a point of reference, eslint let comment ranges bake for a long before adding any sugar for single-line enable/disable, next-line, etc. They don't have a concept of "attached" and as far as I can tell no one has ever asked for it)